modules/snooker/src/main/ts/ephox/snooker/operate/Render.ts

```diff
@@ -47,6 +61,10 @@ const render = (rows: number, columns: number, rowHeaders: number, columnHeaders
 
   const actualRowHeaders = Math.min(rows, rowHeaders);
```

Comment:
This is a bit confusing, as this is related to generating the table heads, but then in between you've added logic for the colgroup. I'd suggest moving this line below inserting the colgroups.